### PR TITLE
Using radixtree for check spatialIDs' overlap

### PR DESCRIPTION
**Pull request by mi-24v**
- using [multidimensional-radix-tree](https://github.com/trajectoryjp/multidimensional-radix-tree) for `CheckSpatialIdsOverlap` and `CheckSpatialIdsArrayOverlap`
  - these functions were moved into `detector` package to prevent cycle import
- new conversion function that adds offset to f-index for radix tree
  - since negative value cannot be inserted this tree

### CODE REVIEW

**Review by HarutakaMatsumoto — CHANGES_REQUESTED** [review]
I reviewed this.
Check it out.

File: test/detect_overlap/main.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/trajectoryjp/spatial_id_go/v4/detector"
+	_ "net/http/pprof"
+	"os"
+	"runtime/pprof"
+	"time"
+)
+
+const maxCases = 10
+
+func main() {
```
[the remark this answers is not in the document]
Comment (mi-24v) [reply test/detect_overlap/main.go]: moved at [d1587dd](https://github.com/trajectoryjp/spatial_id_go/pull/35/commits/d1587dd2528c7f5feb363c4ed47689c61b998f94) and [e6054cc](https://github.com/trajectoryjp/spatial_id_go/pull/35/commits/e6054ccc06ea26cc8f98b05b6c44befc53c1fd92)

File: transform/convert_quadkey_and_Vertical_id.go [hunk 2]
```diff
@@ -1065,19 +1061,96 @@ func convertZToMinAltitudekey(inputIndex int64, inputZoom int64, outputZoom int6
...
+	if inputIndex > maxInputIndex || inputIndex < minInputIndex {
+		return errors.NewSpatialIdError(errors.InputValueErrorCode, "input index does not exist"), false
+	}
+	return nil, true
+}
+
+// AddZBaseOffsetToZ (拡張)空間IDz成分(fインデックス)にオフセット加算する
+// ConvertZToAltitudekey()でズームレベル変換を行わずオフセット変換のみ指定したもの
+//
+// したがって ConvertZToAltitudekey(f, z, z, consts.ZBaseExponent, zBaseOffset)と等価
```
[the remark this answers is not in the document]
Comment (mi-24v) [reply transform/convert_quadkey_and_Vertical_id.go]: changed at [056776c](https://github.com/trajectoryjp/spatial_id_go/pull/35/commits/056776c4a44c233e34b83ca4fb0b0a52e472801d)

And `AddZBaseOffsetToZ` was removed at [a0e7440](https://github.com/trajectoryjp/spatial_id_go/pull/35/commits/a0e744016df071d6e4b05c6e5ade5dcc47a9f70c)

**Comment by HarutakaMatsumoto** [comment]
@mi-24v 
I merged it.
Thank you for your contribution!

**Pull request merged**